eclipse-theia/theia
# option for using a ${targetPlatform} placeholder in theiaPlugins urls for non universal plugins

in plugin download urls



#### What it does
this allows the usage of placeholders in plugin urls for the `theia download:plugins` task. Specificly adds the ${targetPlatform} placeholder to download non universal plugins by the bundling machines operating system and cpu arcitecture.


usage looks like this: 
```
"theiaPlugins": {
   ...
   "rust-analyzer": "https://open-vsx.org/api/rust-lang/rust-analyzer/${targetPlatform}/0.4.1473/file/rust-lang.rust-analyzer-0.4.1473@${targetPlatform}.vsix"
}
```

#### How to test
1. add `"rust-analyzer": "https://open-vsx.org/api/rust-lang/rust-analyzer/${targetPlatform}/0.4.1473/file/rust-lang.rust-analyzer-0.4.1473@${targetPlatform}.vsix"` to your theiaPlugins
2. execute theia download:plugins
3. look at the extension.vsixmanifest from rust-anylzer in the plugins directory and see it should be whatever system your running on.
#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
